src/main/java/Duke.java

```diff
@@ -1,10 +1,100 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    public static String readPrompt(Scanner in){
+        return in.nextLine();
+    }
+/*
+    public static void echoPrompt(String input){
+        System.out.println(input);
+    }
+*/
+
+    public static void displayList(Task[] tasks, int taskIndex){
+        if (taskIndex==0){
+            System.out.println("\tYou haven't noted down anything yet.");
+            return;
+        }
+        System.out.println("\tHere are the tasks in your list:");
+        for (int i=0; i<taskIndex; i++){
+            System.out.println("\t" + (i+1) + "." + tasks[i].toString());
+        }
+    }
+
+    private static void addToList(Task[] tasks, String prompt, int taskIndex, String taskType) {
+        int cutOffPoint = 0;
+        switch(taskType) {
+            case "todo":
+                tasks[taskIndex] = new Todo(prompt);
+                break;
+            case "deadline":
+                cutOffPoint = prompt.indexOf("/by");
+                tasks[taskIndex] = new Deadline(prompt.substring(0,cutOffPoint-1), prompt.substring(cutOffPoint+4));
+                break;
+            case "event":
+                cutOffPoint = prompt.indexOf("/at");
+                tasks[taskIndex] = new Event(prompt.substring(0,cutOffPoint-1), prompt.substring(cutOffPoint+4));
+                break;
+        }
+        System.out.println("\tGot it. I've added this task:\n" +
+                            "\t  " + tasks[taskIndex].toString());
+        System.out.println("\tNow you have "+ (taskIndex+1) +" tasks in the list.");
+    }
+
+    private static void completeTask(Task[] tasks, String substring) {
+        int taskIndex = Integer.parseInt(substring) - 1;
+        tasks[taskIndex].markAsDone();
+        System.out.println("\tNice! I've marked this task as done: \n" +
+                            "\t" + tasks[taskIndex].toString());
+        //bug here: does not deal with overflow values of input
+    }
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
```

Comment:
Consider removing the logo variable can be deleted. (It is not used in the rest of the code)